packages/agent/src/agent.ts

```diff
@@ -22,32 +22,54 @@ export default class Agent {
   private linkerName: string
   private plugins: { [name: string]: any }
 
-  constructor (projectId: string, settings: Object | 'auto') {
+  constructor (
+    projectId: string,
+    cookieDomain: string,
+    fieldsObject?: FieldsObject
+  ) {
     this.plugins = {}
-    if (typeof settings === 'string' && settings === 'auto') {
-      settings = { auto: true }
-    }
-    const config = objectAssign({}, SETTINGS_DEFAULT, settings)
+    const config = objectAssign(
+      {},
+      SETTINGS_DEFAULT,
+      { cookieDomain },
+      fieldsObject || {}
+    ) as SettingFieldsObject
+
     this.core = new AgentCore(
       projectId,
       [Click, MouseMove, Scroll, TouchEnd],
       config
     )
     this.linkerName = config.linkerName
-    setup(config.Raven)
+
+    setup(config)
+
     if (validate(LISTENER.concat(['onpagehide']))) {
       window.addEventListener(
         'pagehide',
         () => {
-          this.core.sendInteracts(true)
+          this.core.sendInteracts([], true)
         },
         false
       )
     }
   }
 
-  send (type: HitType, data?: SendData): AgentCore {
-    this.core.send(type, data || location.href)
+  send (type: HitType, data?: FieldsObject | string): AgentCore {
+    if (typeof type !== 'string') {
+      return this.core
+    }
+    switch (type) {
+      case 'pageview':
+        if (!data || typeof data === 'string') {
```

Comment:
Does not need strict undefined check?

Reply:
`!data` includes `'' ? 'defined' : "undefined"`